pyiceberg/partitioning.py

```diff
@@ -85,6 +91,20 @@ def __str__(self) -> str:
         """Return the string representation of the PartitionField class."""
         return f"{self.field_id}: {self.name}: {self.transform}({self.source_id})"
 
+    def __hash__(self) -> int:
+        """Return the hash of the partition field."""
+        return hash((self.name, self.source_id, self.field_id, repr(self.transform)))
```

Comment:
I should probably just implement `hash` for each of the transforms. The repr is unique but it feels weird to take a hash of a repr.

Reply:
This is a bit awkward indeed. I would leave out the transform since it doesn't carry any information until it is bound.